frontend/src/global_types.ts

```diff
@@ -47,11 +48,32 @@ export type CodeBlock = {
   source: string | null,
 }
 
+export type C2Event = {
+  type: 'c2-event',
+  c2: string,         // "Cobalt Strike"
+  c2Operator: string, // The c2 frameworke user that issued task
+  beacon: string,     // Beacon identifier
+  externalIP: string,
+  internalIP: string,
+  hostname: string,
+  userContext: string,// The user context that the implant/beacon/agent is running under
+  integrity: string,  // "Low", "Medium", "High", "System"
+  processName: string,// process image file shortname
+  processID: number,  // is 'number' acceptable here? Its a float :/
```

Comment:
Yeah, `number` could be fine here. That corresponds to a double sized float. 

But, are you sure this is a number at all? I don't know the format, but usually IDs are strings, even if they're only comprised of numbers. A number implies you can, say, add processIDs. Does it make sense to add these IDs?

Reply:
That being the case, string may be more appropriate.

Reply:
processID here is the PID of the beacon process - for context.